### PR TITLE
Fix pilot-agent application port 0

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
I have noticed such error, `0` is not a port. 

* envoy missing listener for inbound application port: 0
* envoy missing listener for inbound application port: 9080

### CODE REVIEW

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ func getDNSDomain(domain string) string {
 }
 
 func parseApplicationPorts() ([]uint16, error) {
-	parsedPorts := make([]uint16, len(applicationPorts))
+	parsedPorts := []uint16{}
```
Comment (nmittler) [inline pilot/cmd/pilot-agent/main.go]: Actually, I think this should be `make([]uint16, 0, len(applicationPorts))`

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go]: SGTM

**Comment by hzxuzhonghu** [comment]
@wenchenglu please help approve

**Comment by sdake** [comment]
@knrc FYI ^^

**Comment by knrc** [comment]
@sdake shucks I was going to report this, I fixed it locally yesterday while investigating the other issue

**Pull request merged**